packages/components/src/Datalist/Datalist.component.test.js

```diff
@@ -111,15 +111,69 @@ describe('Datalist component', () => {
 			/>,
 		);
 		const input = wrapper.find('input').at(0);
+		input.simulate('change', { target: { value: 'foobar' } });
+		expect(wrapper.find(Typeahead).props().items.length).toBe(1);
+
+		// when
+		input.simulate('blur');
+
+		// then
+		const payload = { value: 'foobar' };
+		expect(onChange).toBeCalledWith(expect.anything(), payload);
+		expect(wrapper.find(Typeahead).props().items).toBe(null);
+	});
+
+	it('should reset suggestions and change value on blur when not in restricted mode', () => {
```

Comment:
This one is the exact same as the previous one, you can remove it.

Reply:
not exactly, this one is to check that the value is selected even if it does not exist in the suggestions